### DISCUSSION TITLE
connect to statemachine loaded from xml-file

### DISCUSSION THREAD

**Discussion by PhilWillecke** [FreeOpcUa/opcua-asyncio#1189] [General]
Hello all,

I am a beginner with OPCUA-asyncio and I am grateful for some support.

I want to build a server that contains a StateMachine. This should behave like a ProgramStateMachine. I already have an xml-file which I import and build my server from (see picture).
Now I want to be able to switch between the states with the methods contained in the StateMachine. Is there a way to connect to the loaded StateMachine?

With the methods this works with link_method so I am looking for something for the StateMachine.

I am grateful for any support or alternative approach.
![image](https://user-images.githubusercontent.com/74958641/215826027-98a020ec-506e-4c02-9fec-c8128a7e06f6.png)

**Comment by AndreasHeine** [comment]
see https://github.com/FreeOpcUa/opcua-asyncio/issues/1149

edit: sorry for the delay atm not much time for non paid opensource work ;)
